From 9e1de0748b65727971ff1783ea5ef66aaa8345e8 Mon Sep 17 00:00:00 2001
From: Xing Zheng <zhengxing@rock-chips.com>
Date: Wed, 30 Sep 2020 17:43:28 +0800
Subject: [PATCH] ASoC: rk3308_codec: fix the hidden ADC MIC PGA Gains

When we name 'Volume' for TLV, the 'Volume' will be ignored
and there are some confilcts name 'ADC MIC Group X Left/Right'.
Therefore, using 'Gain' instead of 'Volume'.

Signed-off-by: Xing Zheng <zhengxing@rock-chips.com>
Change-Id: I5380cdd376dbd69ae50bee7b164a01f954df53f9
---
 sound/soc/codecs/rk3308_codec.c | 32 ++++++++++++++++----------------
 1 file changed, 16 insertions(+), 16 deletions(-)

diff --git a/sound/soc/codecs/rk3308_codec.c b/sound/soc/codecs/rk3308_codec.c
index 03eceaa1058f..f318fc9e274b 100644
--- a/sound/soc/codecs/rk3308_codec.c
+++ b/sound/soc/codecs/rk3308_codec.c
@@ -413,7 +413,7 @@ static const struct soc_enum rk3308_agc_asr_enum_array[] = {
 
 static const struct snd_kcontrol_new mic_gains_a[] = {
 	/* ADC MIC */
-	SOC_SINGLE_EXT_TLV("ADC MIC Group 0 Left Volume",
+	SOC_SINGLE_EXT_TLV("ADC MIC Group 0 Left Gain",
 			   RK3308_ADC_ANA_CON01(0),
 			   RK3308_ADC_CH1_MIC_GAIN_SFT,
 			   RK3308_ADC_CH1_MIC_GAIN_MAX,
@@ -421,7 +421,7 @@ static const struct snd_kcontrol_new mic_gains_a[] = {
 			   rk3308_codec_mic_gain_get,
 			   rk3308_codec_mic_gain_put,
 			   rk3308_codec_adc_mic_gain_tlv_a),
-	SOC_SINGLE_EXT_TLV("ADC MIC Group 0 Right Volume",
+	SOC_SINGLE_EXT_TLV("ADC MIC Group 0 Right Gain",
 			   RK3308_ADC_ANA_CON01(0),
 			   RK3308_ADC_CH2_MIC_GAIN_SFT,
 			   RK3308_ADC_CH2_MIC_GAIN_MAX,
@@ -429,7 +429,7 @@ static const struct snd_kcontrol_new mic_gains_a[] = {
 			   rk3308_codec_mic_gain_get,
 			   rk3308_codec_mic_gain_put,
 			   rk3308_codec_adc_mic_gain_tlv_a),
-	SOC_SINGLE_EXT_TLV("ADC MIC Group 1 Left Volume",
+	SOC_SINGLE_EXT_TLV("ADC MIC Group 1 Left Gain",
 			   RK3308_ADC_ANA_CON01(1),
 			   RK3308_ADC_CH1_MIC_GAIN_SFT,
 			   RK3308_ADC_CH1_MIC_GAIN_MAX,
@@ -437,7 +437,7 @@ static const struct snd_kcontrol_new mic_gains_a[] = {
 			   rk3308_codec_mic_gain_get,
 			   rk3308_codec_mic_gain_put,
 			   rk3308_codec_adc_mic_gain_tlv_a),
-	SOC_SINGLE_EXT_TLV("ADC MIC Group 1 Right Volume",
+	SOC_SINGLE_EXT_TLV("ADC MIC Group 1 Right Gain",
 			   RK3308_ADC_ANA_CON01(1),
 			   RK3308_ADC_CH2_MIC_GAIN_SFT,
 			   RK3308_ADC_CH2_MIC_GAIN_MAX,
@@ -445,7 +445,7 @@ static const struct snd_kcontrol_new mic_gains_a[] = {
 			   rk3308_codec_mic_gain_get,
 			   rk3308_codec_mic_gain_put,
 			   rk3308_codec_adc_mic_gain_tlv_a),
-	SOC_SINGLE_EXT_TLV("ADC MIC Group 2 Left Volume",
+	SOC_SINGLE_EXT_TLV("ADC MIC Group 2 Left Gain",
 			   RK3308_ADC_ANA_CON01(2),
 			   RK3308_ADC_CH1_MIC_GAIN_SFT,
 			   RK3308_ADC_CH1_MIC_GAIN_MAX,
@@ -453,7 +453,7 @@ static const struct snd_kcontrol_new mic_gains_a[] = {
 			   rk3308_codec_mic_gain_get,
 			   rk3308_codec_mic_gain_put,
 			   rk3308_codec_adc_mic_gain_tlv_a),
-	SOC_SINGLE_EXT_TLV("ADC MIC Group 2 Right Volume",
+	SOC_SINGLE_EXT_TLV("ADC MIC Group 2 Right Gain",
 			   RK3308_ADC_ANA_CON01(2),
 			   RK3308_ADC_CH2_MIC_GAIN_SFT,
 			   RK3308_ADC_CH2_MIC_GAIN_MAX,
@@ -461,7 +461,7 @@ static const struct snd_kcontrol_new mic_gains_a[] = {
 			   rk3308_codec_mic_gain_get,
 			   rk3308_codec_mic_gain_put,
 			   rk3308_codec_adc_mic_gain_tlv_a),
-	SOC_SINGLE_EXT_TLV("ADC MIC Group 3 Left Volume",
+	SOC_SINGLE_EXT_TLV("ADC MIC Group 3 Left Gain",
 			   RK3308_ADC_ANA_CON01(3),
 			   RK3308_ADC_CH1_MIC_GAIN_SFT,
 			   RK3308_ADC_CH1_MIC_GAIN_MAX,
@@ -469,7 +469,7 @@ static const struct snd_kcontrol_new mic_gains_a[] = {
 			   rk3308_codec_mic_gain_get,
 			   rk3308_codec_mic_gain_put,
 			   rk3308_codec_adc_mic_gain_tlv_a),
-	SOC_SINGLE_EXT_TLV("ADC MIC Group 3 Right Volume",
+	SOC_SINGLE_EXT_TLV("ADC MIC Group 3 Right Gain",
 			   RK3308_ADC_ANA_CON01(3),
 			   RK3308_ADC_CH2_MIC_GAIN_SFT,
 			   RK3308_ADC_CH2_MIC_GAIN_MAX,
@@ -481,7 +481,7 @@ static const struct snd_kcontrol_new mic_gains_a[] = {
 
 static const struct snd_kcontrol_new mic_gains_b[] = {
 	/* ADC MIC */
-	SOC_SINGLE_EXT_TLV("ADC MIC Group 0 Left Volume",
+	SOC_SINGLE_EXT_TLV("ADC MIC Group 0 Left Gain",
 			   RK3308_ADC_ANA_CON01(0),
 			   RK3308_ADC_CH1_MIC_GAIN_SFT,
 			   RK3308_ADC_CH1_MIC_GAIN_MAX,
@@ -489,7 +489,7 @@ static const struct snd_kcontrol_new mic_gains_b[] = {
 			   rk3308_codec_mic_gain_get,
 			   rk3308_codec_mic_gain_put,
 			   rk3308_codec_adc_mic_gain_tlv_b),
-	SOC_SINGLE_EXT_TLV("ADC MIC Group 0 Right Volume",
+	SOC_SINGLE_EXT_TLV("ADC MIC Group 0 Right Gain",
 			   RK3308_ADC_ANA_CON01(0),
 			   RK3308_ADC_CH2_MIC_GAIN_SFT,
 			   RK3308_ADC_CH2_MIC_GAIN_MAX,
@@ -497,7 +497,7 @@ static const struct snd_kcontrol_new mic_gains_b[] = {
 			   rk3308_codec_mic_gain_get,
 			   rk3308_codec_mic_gain_put,
 			   rk3308_codec_adc_mic_gain_tlv_b),
-	SOC_SINGLE_EXT_TLV("ADC MIC Group 1 Left Volume",
+	SOC_SINGLE_EXT_TLV("ADC MIC Group 1 Left Gain",
 			   RK3308_ADC_ANA_CON01(1),
 			   RK3308_ADC_CH1_MIC_GAIN_SFT,
 			   RK3308_ADC_CH1_MIC_GAIN_MAX,
@@ -505,7 +505,7 @@ static const struct snd_kcontrol_new mic_gains_b[] = {
 			   rk3308_codec_mic_gain_get,
 			   rk3308_codec_mic_gain_put,
 			   rk3308_codec_adc_mic_gain_tlv_b),
-	SOC_SINGLE_EXT_TLV("ADC MIC Group 1 Right Volume",
+	SOC_SINGLE_EXT_TLV("ADC MIC Group 1 Right Gain",
 			   RK3308_ADC_ANA_CON01(1),
 			   RK3308_ADC_CH2_MIC_GAIN_SFT,
 			   RK3308_ADC_CH2_MIC_GAIN_MAX,
@@ -513,7 +513,7 @@ static const struct snd_kcontrol_new mic_gains_b[] = {
 			   rk3308_codec_mic_gain_get,
 			   rk3308_codec_mic_gain_put,
 			   rk3308_codec_adc_mic_gain_tlv_b),
-	SOC_SINGLE_EXT_TLV("ADC MIC Group 2 Left Volume",
+	SOC_SINGLE_EXT_TLV("ADC MIC Group 2 Left Gain",
 			   RK3308_ADC_ANA_CON01(2),
 			   RK3308_ADC_CH1_MIC_GAIN_SFT,
 			   RK3308_ADC_CH1_MIC_GAIN_MAX,
@@ -521,7 +521,7 @@ static const struct snd_kcontrol_new mic_gains_b[] = {
 			   rk3308_codec_mic_gain_get,
 			   rk3308_codec_mic_gain_put,
 			   rk3308_codec_adc_mic_gain_tlv_b),
-	SOC_SINGLE_EXT_TLV("ADC MIC Group 2 Right Volume",
+	SOC_SINGLE_EXT_TLV("ADC MIC Group 2 Right Gain",
 			   RK3308_ADC_ANA_CON01(2),
 			   RK3308_ADC_CH2_MIC_GAIN_SFT,
 			   RK3308_ADC_CH2_MIC_GAIN_MAX,
@@ -529,7 +529,7 @@ static const struct snd_kcontrol_new mic_gains_b[] = {
 			   rk3308_codec_mic_gain_get,
 			   rk3308_codec_mic_gain_put,
 			   rk3308_codec_adc_mic_gain_tlv_b),
-	SOC_SINGLE_EXT_TLV("ADC MIC Group 3 Left Volume",
+	SOC_SINGLE_EXT_TLV("ADC MIC Group 3 Left Gain",
 			   RK3308_ADC_ANA_CON01(3),
 			   RK3308_ADC_CH1_MIC_GAIN_SFT,
 			   RK3308_ADC_CH1_MIC_GAIN_MAX,
@@ -537,7 +537,7 @@ static const struct snd_kcontrol_new mic_gains_b[] = {
 			   rk3308_codec_mic_gain_get,
 			   rk3308_codec_mic_gain_put,
 			   rk3308_codec_adc_mic_gain_tlv_b),
-	SOC_SINGLE_EXT_TLV("ADC MIC Group 3 Right Volume",
+	SOC_SINGLE_EXT_TLV("ADC MIC Group 3 Right Gain",
 			   RK3308_ADC_ANA_CON01(3),
 			   RK3308_ADC_CH2_MIC_GAIN_SFT,
 			   RK3308_ADC_CH2_MIC_GAIN_MAX,
-- 
2.35.3

